packages/block-editor/src/store/actions.js

```diff
@@ -1300,15 +1304,20 @@ export function* insertBeforeBlock( clientId ) {
 		clientId,
 		rootClientId
 	);
+	if ( blockName ) {
+		const block = createBlock( blockName, {} );
+		return yield insertBlock( block, firstSelectedIndex, rootClientId );
+	}
```

Comment:
I was trying to avoid a selector call in an action creator, but I think this works too. I ended up making __experimentalGetDefaultBlockForAllowedBlocks a bit smarter too, so we wouldn't need the caller to fallback to `getDefaultBlockName` in https://github.com/WordPress/gutenberg/pull/29490/commits/7df451e58a37e205c14caa1ed95b1c61ca22eb32